ark/analysis/spatial_analysis.py

```diff
@@ -324,7 +324,14 @@ def calculate_cluster_spatial_enrichment(all_data, dist_matrices_dict, included_
     cluster_num = len(cluster_ids)
 
     # Only include the columns with the patient label, cell label, and cell phenotype
-    all_pheno_data = all_data[[fov_col, cell_label_col, cluster_id_col]]
+    all_pheno_data_cols = [fov_col, cell_label_col, cluster_id_col]
+    all_pheno_data_cols += [] if context_col is None else [context_col]
+
+    all_pheno_data = all_data[all_pheno_data_cols]
+
+    if context_col is not None:
+        context_names = all_data[context_col].unique()
+        context_pairings = combinations_with_replacement(context_names, 2)
```

Comment:
Nice!

Reply:
How does this handle empty rows? I assume np.unique() will treat empty as a unique value?

Lets say the output from cell clustering is 20 different cell types. There will be one column where each row (i.e. cell) is assigned to one of those possible cell types. For context-dependent randomization, we might only be interested in the interactions between the 7 different immune celltypes that are there. We could make a separate column that only includes the immune celltypes, but then the other cell types won't be labeled. 

The other option would be to pass in a list of values that we want to context randomize over, but that could also get messy.

Reply:
Actually I rescind my comment. The way this context randomization was originally done by Leeat wasn't just for one specific subtype, but across a couple different celltypes. I think this is a good way to set things up for now.

Reply:
In the futre, we can add a 'context filter' to filter cell tables to only compute spatial enrichment between cells in the given list of contexts.  That's a simple as adding this line:
```
filtered_cell_table = cell_table[cell_table[context_col].isin(contexts)]
```